AutomaticQC/SpikeClassifiers/imosSpikeClassifierBurstHampel.m

```diff
@@ -0,0 +1,176 @@
+function [spikes] = imosSpikeClassifierBurstHampel(bindexes, signal, use_burst_window, half_window_width, madfactor, lower_mad_limit, repeated_only)
+% function [spikes] = imosSpikeClassifierBurstHampel(bindexes, signal,  use_burst_window,  half_window_width, madfactor, lower_mad_limit, repeated_only)
+%
+% A Hampel classifier for burst data. The half_window_width is applied at burst level instead of
+% sample level.
+%
+% Inputs:
+%
+% bindexes - a cell with the invidivudal burst range indexes.
+% signal - A Burst 1-d signal.
+% use_burst_window  - a boolean to consider the half_window_width applied at burst scale. If false, will consider the burst series as continuous.
+% half_window_width - The half_window_width burst range (the full window size will be 1+2*half_window_width)
+% madfactor -  A multipling scale for the MAD.
+% repeated_only - a boolean to mark spikes in burst only if they are detected more than one time (only for half_window_width>0).
+% lower_mad_limit - a lower threshold for the MAD values, which values below will be ignored.
+%
+% Outputs:
+%
+% spikes - An array with spikes indexes.
```

Comment:
when i look at the example and implementation, to me spikes is an array of spike values from the signal, not indexes. Am i missing something out?

Reply:
the spikes output are always indexes where spike occur, not the actual value of the spike. You should not find anywhere the [spikes] output returning values, only indexes.

Reply:
ok i really don't understand what is happening then

you have in your example
fullburst_spiked = [3,7,13,14,15,16,17,18,33,92,99]
...
assert(isequal(dspikes,[3,7,33,92,99]));

to me the values in the assert equal are pretty similar to the fullburst_spiked array, and not indexes

Reply:
fullburst_spiked defined above is an array of indexes where the spikes are found. Just below this definition we got:
```
x(fullburst_spiked) = 1000;
```
Hence, these are the index locatiosn where values are 1000 in the random vector x.

The assertion you mention is correct. The function is configured to use the entire burst as a window. hence, It's missing the ```13,14,15,16,17,18``` because these are *all* the positions of a single burst (burst length here in the example is equal to 6). Hence, the entire burst is marked with value 1000. Thus, as configure per defaults, the burst statistics are mean/median == 1000 and std ==0; no spikes will be detected by definition. However, just below, if one seeks to detect those "entire bursts as a spike", you need to change the args to use more than one burst within a window. When this is the case, an entire burst with spiked data can be detected as per docstring.